Teleport Install scripts: use `/etc/apt/keyrings/` when adding Teleport's public key for DEB packages

marcoandredinis opened the pull request:
Same as https://github.com/gravitational/teleport/pull/50033

Demo:
```
root@cd3db2db3032:/# cat /etc/apt/sources.list.d/teleport.list
deb [signed-by=/etc/apt/trusted.gpg.d/teleport-archive-keyring.asc]             https://apt.releases.teleport.dev/ubuntu noble stable/v17
root@cd3db2db3032:/# cat /etc/apt/trusted.gpg.d/teleport-archive-keyring.asc | head
-----BEGIN PGP PUBLIC KEY BLOCK-----

mQINBF+R3LYBEADOEO9i3Dm5rEAiXONchX3M54QzZX0yHArSpYQ5aJDdJRQbqzqT
+e2os8NpSjVDZFNz5ul8xkZsnCLX7pgrAYqq+vsXL4bMWDP96S6PjfVIAyV4ylv0
DBReMdkaAZb/IoPhkSTT+ayw4eGEtUz/k7mxMpQ9ob7qFtGs8aNVT/An5LfFR1Lx
9WOlFPPIAJKcHVIyRD+4EoCSn1R1c61UHFIRatbAnwOLs3iz4/GU+w9wdbuWbDuk
nGdG0Lmlzp42HHxeJJFQlOTed97+trktvAiuzA/0lbQHEcWvxfWAy5//cjORp+H3
RGLp8fJ+fFRAyA4WP6O3wIC4gAAgsEn8WpVT8wZYlLMRf694SeawBtyUSlcsn9i1
LuOh5akOY3iQtH01+rMBjOaMkCmpT2nQaUH+HS2iZBddBHdAMMQtj2UolMRbUSxH
+GJczes1t9/WH3vbvh5ESMOy0fH14Tjo+9yQYa4EhFNNloAG10DYFLlCj47fWDdS
root@cd3db2db3032:/# apt list teleport -a
Listing... Done
teleport/noble,now 17.0.5 arm64 [installed]
teleport/noble 17.0.4 arm64
teleport/noble 17.0.3 arm64
teleport/noble 17.0.2 arm64
teleport/noble 17.0.1 arm64
```

marcoandredinis committed:
Teleport Install scripts: use `/etc/apt/trusted.gpg.d/` for DEB packages

gzdunek reviewed (commented):
I don't have a full understanding of this topic, but here they say that adding keys to `/etc/apt/trusted.gpg.d/` is insecure https://stackoverflow.com/a/71384057.
What do you think of it?

marcoandredinis committed:
use etc apt keyrings for storing the keys

marcoandredinis commented:
> I don't have a full understanding of this topic, but here they say that adding keys to `/etc/apt/trusted.gpg.d/` is insecure https://stackoverflow.com/a/71384057. What do you think of it?

I've changed the destination folder to be `etc/apt/keyrings`
I've also added a comment in the original PR.

Thank you for bringing this up. I [tried to explain](https://github.com/gravitational/teleport/pull/50033#issuecomment-2535807069) the downsides of this approach but didn't actually came up with a better solution.

gzdunek reviewed (approved):
Thanks for looking into this, we should be safer now.
